bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/handler/EmbyBridgeHandler.java

```diff
@@ -0,0 +1,277 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.emby.internal.handler;
+
+import static java.util.Objects.requireNonNull;
+import static org.openhab.binding.emby.internal.EmbyBindingConstants.CONNECTION_CHECK_INTERVAL_MS;
+
+import java.util.Collections;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.openhab.binding.emby.internal.EmbyBridgeConfiguration;
+import org.openhab.binding.emby.internal.EmbyBridgeListener;
+import org.openhab.binding.emby.internal.discovery.EmbyClientDiscoveryService;
+import org.openhab.binding.emby.internal.model.EmbyPlayStateModel;
+import org.openhab.binding.emby.internal.protocol.EmbyConnection;
+import org.openhab.binding.emby.internal.protocol.EmbyHTTPUtils;
+import org.openhab.binding.emby.internal.protocol.EmbyHttpRetryExceeded;
+import org.openhab.core.config.core.validation.ConfigValidationException;
+import org.openhab.core.config.core.validation.ConfigValidationMessage;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EmbyBridgeHandler} is responsible for handling commands,
+ * managing the connection to Emby and dispatching play‐state events.
+ * 
+ * @author Zachary Christiansen - Initial Contribution
+ */
+@NonNullByDefault
+public class EmbyBridgeHandler extends BaseBridgeHandler implements EmbyBridgeListener {
+
+    private final Logger logger = LoggerFactory.getLogger(EmbyBridgeHandler.class);
+
+    private volatile @Nullable EmbyConnection connection;
+    private final WebSocketClient webSocketClient;
+    private @Nullable ScheduledFuture<?> connectionCheckerFuture;
+    private @Nullable EmbyClientDiscoveryService clientDiscoveryService;
+    private @Nullable EmbyHTTPUtils httputils;
+    private @Nullable EmbyBridgeConfiguration config;
+    private int reconnectionCount;
+    private @Nullable String lastDiscoveryStatus;
+    @Reference
+    private @NonNullByDefault({}) TranslationProvider i18nProvider;
```

Comment:
Please inject this in the constructor / factory similar to the `WebSocketFactory`

Reply:
changed for both the bridge handler and the device handler